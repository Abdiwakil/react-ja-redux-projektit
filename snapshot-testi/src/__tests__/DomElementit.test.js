import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import DomElementit from '../DomElementit';
import { create } from 'react-test-renderer';
 
afterEach(cleanup);
 
it('pitäisi olla 0', () => {
    const { getByTestId } = render(<DomElementit />);
    expect(getByTestId('laskuri')).toHaveTextContent(0);
});
 
it('pitäisi olla enabled', ()=> {
    const { getByTestId } = render(<DomElementit />);
    expect(getByTestId('nappi-ylos')).not.toHaveAttribute('disabled');
})
 
it('pitaisi olla disabled', ()=> {
    const { getByTestId } = render(<DomElementit />);
    expect(getByTestId('nappi-alas')).toBeDisabled();
})
describe('Snapshot-testi', () => {
    test('testataan DomElementit', () => {
        let tree = create(<DomElementit />)
        expect(tree.toJSON()).toMatchSnapshot();
    })
})
