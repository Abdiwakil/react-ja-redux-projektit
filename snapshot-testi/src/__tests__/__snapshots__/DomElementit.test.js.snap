// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot-testi testataan DomElementit 1`] = `
Array [
  <h1
    data-testid="laskuri"
  >
    0
  </h1>,
  <button
    data-testid="nappi-ylos"
    onClick={[Function]}
  >
    Yl√∂s
  </button>,
  <button
    data-testid="nappi-alas"
    disabled={true}
    onClick={[Function]}
  >
    Alas
  </button>,
]
`;
